### DISCUSSION TITLE
API-style suited?

### DISCUSSION THREAD

**Discussion by ddahan** [withlogicco/django-prose#30] [Q&A]
Hi guys!
Very hyped to hear about this project :) Can't wait to try it.

I'm currently using simple Django `TextField` fields to handle content. I put some markdown in them.
Then, I return this markdown using my Django API, and let my front-end (Vue.js) displays it correctly, in two steps:
1. A client lib is used to convert markdown to HTML equivalent
2. [Tailwind Typography plugin](https://tailwindcss.com/docs/typography-plugin) is used to apply some default styling to the HTML content.

It kind of works but is not perfect. From what I understand, using Django-prose, I could win:
- A rich Editor directly in the admin (this is easier than markdown, especially for non tech users)
- Adding images directly when editing content (this is something I definitely need for blogging part!)
(performance and security are not an issue for now).

Questions:
- Do you think Django-prose is suited to my need, even if I don't use Django templates? (Django is used for back-end only)
- if I return a `RichTextField` content, will it contain Markdown or HTML?

Thanks! Keep up the great work.

**Comment by parisk** [comment]
Hey @ddahan,

Great points and questions.

Django Prose is real great to build a blog that you can edit right from the admin panel. Just make sure to use the `Document` model, or a subclass tailored to your needs for the article body! This is better for your database, comparing to keeping the whole body using the `RichTextField`.

Considering the questions:

> Do you think Django-prose is suited to my need, even if I don't use Django templates? (Django is used for back-end only)

You bet! We do as well with a client. That company owns kiosk-like machines, which communicate with an API we have built, fetch the rich text content in HTML format from an endpoint and then display it. 

> if I return a RichTextField content, will it contain Markdown or HTML?

A `RichTextField` contains HTML. So returning it as part of a Django REST Framework serializer for example, will return an HTML string.

Let me know if the above helped and of course, thanks a lot for the kind words!
